unison-cli/src/Unison/Cli/Monad.hs

```diff
@@ -170,18 +175,19 @@ data Env = Env
     sandboxedRuntime :: Runtime Symbol,
     nativeRuntime :: Runtime Symbol,
     serverBaseUrl :: Maybe Server.BaseUrl,
-    ucmVersion :: UCMVersion
+    ucmVersion :: UCMVersion,
+    -- | Whether we're running in a transcript test or not.
+    -- Avoid using this except when absolutely necessary.
+    isTranscriptTest :: Bool
```

Comment:
Added this so that reflog tests weren't flaky, since those say how long ago something happened.